src/lib/optimizer/strategy/between_composition_rule.cpp

```diff
@@ -0,0 +1,284 @@
+
+/**
+ * _get_boundary takes a BinaryPredicateExpression and the corresponding PredicateNode
+ * as its input and returns a normalized ColumnBoundary. This function checks where the
+ * LQPColumnExpression and where the ValueExpression is stored in the BinaryPredicateExpression.
+ * The expressions are transferred to a normalized ColumnBoundary format
+ * and are labled with a ColumnBoundaryType, that depends on their positions and the predicate condition
+ * of the BinaryPredicateExpression
+ *
+ **/
+const BetweenCompositionRule::ColumnBoundary BetweenCompositionRule::_get_boundary(
+    const std::shared_ptr<BinaryPredicateExpression>& expression) const {
+  auto type = ColumnBoundaryType::None;
+  auto column_expression = std::dynamic_pointer_cast<LQPColumnExpression>(expression->left_operand());
+  auto value_expression = std::dynamic_pointer_cast<ValueExpression>(expression->right_operand());
+
+  // Case: "ColumnExpression [CONDITION] ValueExpression" will be checked
+  // Boundary type will be set according to this order
+  if (column_expression != nullptr && value_expression != nullptr) {
+    switch (expression->predicate_condition) {
+      case PredicateCondition::LessThanEquals:
+        type = ColumnBoundaryType::UpperBoundaryInclusive;
+        break;
+      case PredicateCondition::GreaterThanEquals:
+        type = ColumnBoundaryType::LowerBoundaryInclusive;
+        break;
+      case PredicateCondition::LessThan:
+        type = ColumnBoundaryType::UpperBoundaryExclusive;
+        break;
+      case PredicateCondition::GreaterThan:
+        type = ColumnBoundaryType::LowerBoundaryExclusive;
+        break;
+      default:
+        break;
+    }
+  } else {
+    value_expression = std::dynamic_pointer_cast<ValueExpression>(expression->left_operand());
+    column_expression = std::dynamic_pointer_cast<LQPColumnExpression>(expression->right_operand());
+
+    // Case: "ValueExpression [CONDITION] ColumnExpression" will be checked
+    // Boundary type will be set according to this order
+    if (value_expression != nullptr && column_expression != nullptr) {
+      switch (expression->predicate_condition) {
+        case PredicateCondition::GreaterThanEquals:
+          type = ColumnBoundaryType::UpperBoundaryInclusive;
+          break;
+        case PredicateCondition::LessThanEquals:
+          type = ColumnBoundaryType::LowerBoundaryInclusive;
+          break;
+        case PredicateCondition::GreaterThan:
+          type = ColumnBoundaryType::UpperBoundaryExclusive;
+          break;
+        case PredicateCondition::LessThan:
+          type = ColumnBoundaryType::LowerBoundaryExclusive;
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  return {
+      column_expression,
+      value_expression,
+      type,
+  };
+}
+
+static PredicateCondition get_between_predicate_condition(bool left_inclusive, bool right_inclusive) {
```

Comment:
Could we bring the free-floating functions that have to do with handling the PredicateConditions together in one place?